src/wp-admin/includes/dashboard.php

```diff
@@ -1555,3 +1558,251 @@ function wp_welcome_panel() {
 </div>
 	<?php
 }
+
+/**
+ * Returns HTML Mark up for the Petitions Dashboard Widget
+ *
+ * @since 1.0.0
+ */
+function cp_dashboard_petitions_table_body ( $votesCount, $link, $title, $author, $status, $createdTime ) {
+?>
+	<tr>
+		<td class="votes-count"><?php echo esc_attr( $votesCount ); ?></td>
+
+		<td class="petition">
+			<a target="_blank" href="<?php echo esc_url( $link ) ?>"><strong><?php echo esc_attr__( $title )?><span class="screen-reader-text"><?php echo esc_attr__( '(opens in a new window)' ); ?></span><span aria-hidden="true" class="dashicons dashicons-external"></span></strong></a>
+			<?php
+				esc_attr__( 'by' ) . ' ' . ucwords(  esc_attr( $author ) );
+
+				if ( $status == "open" ){
+					echo esc_attr__( ' - ' ) . ' ' . human_time_diff( strtotime( $createdTime ), current_time('timestamp') ) . ' ' . esc_attr__( 'ago' );
+				} 
+				elseif ( $status == "planned" ){
+					echo ' - ' . '<span class="planned">' . esc_attr__( ucfirst( $status ) ) . '</span>';
+				} 
+				else{
+					echo ' - ' . '<span class="started">' . esc_attr__( ucfirst( $status ) ) . '</span>';
+				}
+			?>
+		</td>
+	</tr>
+<?php
+}
+	
+/**
+ * Callback function for the petitions dashboard widget
+ *
+ * @since 1.0.0
+ */
+function cp_dashboard_petitions() {
+
+	// Get any existing copy of our transient data
+	if ( false === ( $dashboard_petitions_results = get_transient( 'dashboard_petitions_results' ) ) ) {
+		/**
+		 * It wasn't there, so regenerate the data and save the transient
+		 * Query API for JSON data -> decode results to php
+		 *
+		 * @since 1.0.0
+		 * @return array
+		 */
+		$api_url = 'https://api-v1.classicpress.net/features/1.0/';
+
+		/**
+		 * Response should be an array with:
+		 *  'most-wanted []' - array() - A user-friendly platform name, if it can be determined
+		 *  'trending []' - array() - A user-friendly browser name
+		 *  'recent []' - array() - The version of the browser the user is using
+		 *  'tags [] - array() - The version of the browser the user is using
+		 *  'link - string - The version of the browser the user is using
+		 */
+		$response = wp_remote_get( $api_url );
+
+		if ( ! is_array( $response ) ) {
+			return;
+		}
```

Comment:
How about a more explicit error check here, something like this:

```
if ( is_wp_error( $response ) || 200 !==  wp_remote_retrieve_response_code( $response ) ) {
    return;
}
```

Reply:
Should we add a UX friendly line? Some thing about broken data?

Reply:
Sure, good idea.  This will come up when the server doesn't have a working internet connection, for example.

We should probably also set a timeout of around 5 seconds (at most) for this request.